src/online/online-audio-source.cc

```diff
@@ -177,12 +177,14 @@ bool OnlineVectorSource::Read(Vector<BaseFloat> *data) {
     if (data->Dim() == subsrc.Dim()) {
       data->CopyFromVec(subsrc);
     } else {
+      data->Resize(n_elem);
```

Comment:
@nshmyrev  @hhadian  can you guys please check this?  I'm suspicious because it looks to me like whoever wrote this code intended that `data` should not be resized.  Incidentally, in this kind of situation you should pass in a pointer to  SubVector, not Vector.  (However, I do think this code has another issue, which is that it fails to zero the unset part of *data.)

Reply:
Hi

I have checked the issue. The only place where it is used is [OnlineFeatureInput](https://github.com/kaldi-asr/kaldi/blob/5968b4cc03f9deccfd566962d3bba96bad8ce522/src/online/online-feat-input.h#L325) and data->Resize is not harmful there. Moreover, some implementations like GstOnlineSource [already resize](https://github.com/kaldi-asr/kaldi/blob/5968b4cc03f9deccfd566962d3bba96bad8ce522/src/gst-plugin/gst-audio-source.cc#L110). TcpInputSource [don't resize](https://github.com/kaldi-asr/kaldi/blob/5968b4cc03f9deccfd566962d3bba96bad8ce522/src/online/online-tcp-source.cc#L143).

Logically I'd better resize than add zero.

So I would:

1. Keep this change as is
2. Add a note on the method documentation comment that the data can be resized.
3. Fix TcpInputSource too.